### ISSUE TITLE
Incompatibility with the latest revision format for Flux

### ISSUE THREAD

**Issue by bpesics** [Azure/gitops-connector#47]
The source revision format reported by the Flux controllers has changed according to [RFC-0005](https://github.com/fluxcd/flux2/tree/main/rfcs/0005-artifact-revision-and-digest)

https://github.com/fluxcd/notification-controller/blob/v0.32.1/CHANGELOG.md#0320

```
DEBUG:urllib3.connectionpool:https://api.github.com:443 "GET /repos/myorg/myrepo/commits/dev@sha1:1fde4efca7a513638ba8f1c836b37f1661e0abd3/status HTTP/1.1" 404 None
ERROR:gitops_event_handler:Exception on /gitopsphase [POST]
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2077, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1525, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1523, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1509, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
  File "/app/gitops_event_handler.py", line 33, in gitopsphase
    gitops_connector.process_gitops_phase(payload, req_time)
  File "/app/gitops_connector.py", line 30, in process_gitops_phase
    if not self._git_repository.is_commit_finished(commit_id):
  File "/app/repositories/github_git_repository.py", line 40, in is_commit_finished
    response.raise_for_status()
  File "/usr/local/lib/python3.9/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://api.github.com/repos/myorg/myrepo/commits/dev@sha1:1fde4efca7a513638ba8f1c836b37f1551e0abd3/status
```

while the correct URL would be something like`https://api.github.com/repos/myorg/myrepo/commits/1fde4efca7a513638ba8f1c836b37f1551e0abd3/status`

**Comment by bpesics** [comment]
https://github.com/Azure/gitops-connector/pull/48

**Comment by vmartinlantek** [comment]
Hey! I've tested the fix implemented by @bigkevmcd and I can confim that it works like a charm for Azure DevOps as well. Could we pelase get this merged and release a new version of the chart?

**Comment by eedorenko** [comment]
Fixed by #49
